### ISSUE TITLE
the result of squad

### ISSUE THREAD

**Issue by TingFree** [studio-ousia/luke#56]
I use poetry to build experiment environment, and I try to reproduce your paper performance follow your advise, bug failed.
                                   EM          F1
paper                         90.2      95.4
your checkpoint        89.76    94.97
your finetune            89.04    94.69

do you know the reason?

**Comment by TingFree** [comment]
i use a single V100 and not use fp16

**Comment by ikuyamada** [comment]
Hi, 
> your checkpoint 89.76 94.97

Did you use the development set, right? These numbers are the same as the reported results on the development set.

**Comment by TingFree** [comment]
Yes, I use the devset to reproduce. What dataset do you use?

**Comment by ikuyamada** [comment]
Actually, the performance you mentioned in this issue (89.76 EM  94.97 F1) is exactly the same as that reported in the paper. Please see the "dev" columns instead of test columns since you use development set.

**Comment by TingFree** [comment]
Sorry, it's my fault, I have not read your paper yet. But I have another question, how do I get your result without your checkpoint? I want to make some improvements based on your model, but I even can't  reproduce your result on dev set with the finetune code your offered.

**Comment by ikuyamada** [comment]
The checkpoint file is trained using the computational environment and hyper-parameters reported in the paper. Unfortunately, since the performance of neural network models changes easily depending on the environment, it is impossible to provide code that produces the same performance on different environments.

**Comment by TingFree** [comment]
Thanks for your reply, I will try to run your model on my surver with your hyper-parameter settings.

**Comment by TingFree** [comment]
Sorry to bother you again, I read your paper, and find it offers the pretrain hyper-parameters and the same hyper-parameters with roberta in SQuAD. I think you set these in code inside, right? You can run the "fine-tuning the model" code you offered in your environment to reproduce the best performance, right? I run your "fine-tuning the model" code, but just 89.04 94.69, is this normal?

**Comment by ikuyamada** [comment]
I think the performance is low. I ran the fine-tuning again using the code publicized on the repository last night, and achieved the same performance as that reported in the paper.

**Comment by TingFree** [comment]
I only use a single v100 and no fp16, is this the problem?

**Comment by ikuyamada** [comment]
I am sorry but I do not know what is the problem. I think it should be possible to reproduce the results based on a single v100 gpu. I confirmed that the code worked properly and achieved the reported performance.

**Comment by TingFree** [comment]
ok, I will check it again, thank you for your patience to reply

**Comment by ikuyamada** [comment]
You may have already noticed, but when you use a single GPU to train the model, you need to change `gradient-accumulation-steps` from 3 to 24. The effective batch size (48 for SQuAD) can be computed by `train_batch_size * gradient_accumulation_steps * num_gpus`.

**Comment by TingFree** [comment]
Indeed I was ignored, what I tried tuning the batch is up to 18, not 48. I'm a novice, Thanks  for reminding me, I run the new parameter now.

**Comment by TingFree** [comment]
I just change the batch size to 48, in a single v100, the performance is 88.95, 94.62, and in a single 1080Ti is 89.0, 94.76, still low

**Comment by ikuyamada** [comment]
Sonce the code achieves the same reported performance in my environment, I believe that there is some technical issue related to your environmemt. I am sorry but I cannot help anymore about this.
